cloud.foundry.cli/src/main/java/cloud/foundry/cli/services/DumpCommandBase.java

```diff
@@ -0,0 +1,51 @@
+package cloud.foundry.cli.services;
+
+import static picocli.CommandLine.Mixin;
+
+import cloud.foundry.cli.crosscutting.logging.Log;
+import cloud.foundry.cli.crosscutting.mapping.YamlMapper;
+import cloud.foundry.cli.crosscutting.mapping.beans.Bean;
+
+import java.util.concurrent.Callable;
+
+/**
+ * The base class for concrete dump commands. It implements the process of resolving and interpreting the provided
+ * configuration file via a {@link Callable<Integer>} interface (as any other command class).
+ * @param <B> the bean type in which the resolved config will be interpreted
+ */
+public abstract class DumpCommandBase<B extends Bean> implements Callable<Integer> {
+
+    @Mixin
+    private YamlCommandOptions yamlCommandOptions;
+
+    private Class<B> beanClass;
+    private Log log;
+
+    /**
+     * Constructs the command instance by the class object of the bean type, that is used in this class, and the class
+     * object of the concrete derived class. This is needed for retrieving the according {@link Log log} instance.
+     * @param beanClass the class object of the bean type
+     * @param commandClass the class object of the dump command class that derived from this class
+     * @param <C> the type of the dump command class that derived from this class
+     */
+    protected <C extends DumpCommandBase<B>> DumpCommandBase(Class<B> beanClass, Class<C> commandClass) {
+        this.beanClass = beanClass;
+        log = Log.getLog(commandClass);
+    }
+
+    /**
+     * Implementation of the dump command functionality. It reads the provided configuration file, resolves
+     * ref-occurrences, prints the resolved configuration and tries to interpret it as the bean type, that is specified
+     * in the generic type of this class.
+     */
+    @Override
+    public Integer call() throws Exception {
+        log.info("Reading and processing YAML file...");
+        String yamlFilePath = yamlCommandOptions.getYamlFilePath();
+        String resolvedConfig = YamlMapper.resolveYamlFile(yamlFilePath);
+        System.out.println(resolvedConfig);
+        YamlMapper.interpretBean(resolvedConfig, beanClass);
```

Comment:
I'm confused. Why print it out before this call?

Reply:
Because the call to interpretBean could end up in an exception. In that case, the user would not get any information about his resolved config, because the program would terminate due to the thrown exception. By first printing the resolved config and afterwards checking the interpretability, we can make sure that the user can actually see the resolved config in any case.

Reply:
So you basically interpret to check the configuration for errors?

Reply:
Yes, the interpretation fails if there are properties in the configuration, that do not fit the properties in the corresponding Bean class